pymc3/distributions/transforms.py

```diff
@@ -265,8 +263,8 @@ class Interval(ElemwiseTransform):
     def __init__(self, a, b):
         self.a = tt.as_tensor_variable(a)
         self.b = tt.as_tensor_variable(b)
-        self.a_ = a
-        self.b_ = b
+        self.a_ = self.a.eval()
```

Comment:
This would break if `a` is a random variable  with upstream dependence right?

Reply:
maybe that's why I am seeing a lot of errors in the tests :-) I will check it ASAP

Reply:
Yeah generally `{tensor}.eval()` is not very reliable as most of the case they have some dependencies.

Reply:
I tried a couple of alternatives, this is the best one. Still a couple of errors. Any suggestion?

Reply:
draw_value would be how I approach it as well, but if `a` is a random variable you will still have problem with `point=None`. Maybe @lucianopaz has some ideas.